stdlib/LinearAlgebra/src/qr.jl

```diff
@@ -542,6 +542,35 @@ function rank(A::QRPivoted; atol::Real=0, rtol::Real=min(size(A)...) * eps(real(
     return something(findfirst(i -> abs(A.R[i,i]) <= tol, 1:m), m+1) - 1
 end
 
+function nullspace(A::QRPivoted; atol::Real=0, rtol::Real=min(size(A)...) * eps(real(float(one(eltype(A.Q))))) * iszero(atol))
+    m, n = size(A, 1), size(A, 2)
+    (m == 0 || n == 0) && return Matrix{eigtype(eltype(A.Q))}(I, n, n)
+
+    indstart = rank(A) + 1
+    return copy((@view collect(A.Q)[:, indstart:end]))
```

Comment:
If I'm not mistaken, this could be rewritten as

```suggestion
    return A.Q[:, indstart:end]
```

It won't first create the whole matrix and then slice it, but it will create a "chopped" identity matrix and then apply `A.Q` to it. That should be actually faster, and save the intermediate full matrix representation of `A.Q`.

Reply:
I wasn't aware this was possible, thank you!